storage/innobase/handler/ha_innodb.cc

```diff
@@ -21839,12 +21839,20 @@ static void update_innodb_redo_log_encrypt(THD *thd, SYS_VAR *var,
     return;
   }
 
-  if (srv_redo_log_encrypt != REDO_LOG_ENCRYPT_OFF &&
-      srv_redo_log_encrypt != target) {
-    push_warning_printf(thd, Sql_condition::SL_WARNING, ER_WRONG_ARGUMENTS,
-                        " Redo log encryption mode"
-                        " can't be switched without stopping the server and"
-                        " recreating the redo logs.");
+  if (existing_redo_encryption_mode != REDO_LOG_ENCRYPT_OFF &&
+      existing_redo_encryption_mode != target) {
+    push_warning_printf(
```

Comment:
may be use ib_senderrf(thd, IB_LOG_LEVEL_WARN, ER_REDO_ENCRYPTION_CANT_BE_CHANGED, log_encrypt_name(existing_redo_encryption_mode), log_encrypt_name(static_cast<redo_log_encrypt_enum>(target))

Reply:
ib::warn() could be unnecessary then? please check